share/rpcauth/rpcauth.py

```diff
@@ -9,26 +9,36 @@
 import base64
 import hmac
 
-if len(sys.argv) < 2:
-    sys.stderr.write('Please include username as an argument.\n')
-    sys.exit(0)
+def generate_salt():
+    # This uses os.urandom() underneath
+    cryptogen = SystemRandom()
 
-username = sys.argv[1]
+    # Create 16 byte hex salt
+    salt_sequence = [cryptogen.randrange(256) for i in range(16)]
+    return ''.join([format(r, 'x') for r in salt_sequence])
 
-#This uses os.urandom() underneath
-cryptogen = SystemRandom()
+def generate_password(salt):
+    # Create 32 byte b64 password
+    password = base64.urlsafe_b64encode(os.urandom(32)).decode('utf-8')
 
-#Create 16 byte hex salt
-salt_sequence = [cryptogen.randrange(256) for i in range(16)]
-hexseq = list(map(hex, salt_sequence))
-salt = "".join([x[2:] for x in hexseq])
+    m = hmac.new(bytearray(salt, 'utf-8'), bytearray(password, 'utf-8'), 'SHA256')
+    password_hmac = m.hexdigest()
 
-#Create 32 byte b64 password
-password = base64.urlsafe_b64encode(os.urandom(32)).decode("utf-8")
+    return (password, password_hmac)
 
-m = hmac.new(bytearray(salt, 'utf-8'), bytearray(password, 'utf-8'), "SHA256")
-result = m.hexdigest()
+def main():
+    if len(sys.argv) < 2:
+        sys.stderr.write('Please include username as an argument.\n')
+        sys.exit(0)
 
-print("String to be appended to bitcoin.conf:")
-print("rpcauth="+username+":"+salt+"$"+result)
-print("Your password:\n"+password)
+    username = sys.argv[1]
+
+    salt = generate_salt()
+    (password, password_hmac) = generate_password(salt)
+
+    print('String to be appended to bitcoin.conf:')
+    print('rpcauth='+username+':'+salt+'$'+password_hmac)
```

Comment:
nit: prefer string formatters (`{}` or `%` - we can't use f-strings yet).

If you must concatenate strings, please surround the `+` with spaces so the code isn't so difficult to parse.

Reply:
Done